specification/http/1.0/openapi.json

```diff
@@ -0,0 +1,487 @@
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/ValidationFailedResponse"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "components": {
+    "securitySchemes": {
+      "bearerAuth": {
+        "type": "http",
+        "scheme": "bearer",
+        "bearerFormat": "JWT"
+      }
+    },
+    "schemas": {
+      "ToolId": {
+        "type": "string",
+        "description": "A tool's unique identifier in the format 'Toolkit.Tool[@version]', where @version is optional.",
+        "pattern": "^[A-Za-z0-9_]+\\.[A-Za-z0-9_]+(@([0-9]+|[0-9]+\\.[0-9]+\\.[0-9]+))?$",
+        "example": [
+          "Calculator.Add",
+          "Calculator.Add@1",
+          "Calculator.Add@1.0.0"
+        ]
+      },
+      "ToolVersion": {
+        "type": "string",
+        "description": "A tool's semantic version in the format 'x.y.z', where x, y, and z are integers.",
+        "pattern": "^[0-9]+\\.[0-9]+\\.[0-9]+$",
+        "example": "1.2.3"
+      },
+      "ToolSecret": {
+        "type": "object",
+        "properties": {
+          "id": {
+            "type": "string",
+            "description": "The secret's unique identifier."
+          },
+          "value": {
+            "type": "string",
+            "description": "The secret's value."
+          }
+        },
+        "required": ["id", "value"],
+        "additionalProperties": false
+      },
+      "ToolDefinition": {
+        "type": "object",
+        "properties": {
+          "id": {
+            "$ref": "#/components/schemas/ToolId"
+          },
+          "name": {
+            "type": "string",
+            "description": "The tool's name. Only allows alphanumeric characters, underscores, and dashes.",
+            "pattern": "^[A-Za-z0-9_-]{1,64}$",
+            "example": ["Calculator_Add", "Email_GetEmails"]
+          },
+          "description": {
+            "type": "string",
+            "description": "A human-readable explanation of the tool's purpose. This field can be used by both humans and AI models."
+          },
+          "version": {
+            "$ref": "#/components/schemas/ToolVersion"
+          },
+          "input_schema": {
+            "type": "object",
+            "properties": {
+              "parameters": {
+                "type": "object",
+                "additionalProperties": true,
+                "description": "JSON Schema describing the input parameters for the tool. Supports standard JSON Schema validation but excludes $ref and definitions/schemas for simplicity."
+              }
+            },
+            "required": ["parameters"],
+            "additionalProperties": false
+          },
+          "output_schema": {
+            "oneOf": [
+              {
+                "type": "object",
```

Comment:
Same comment as above on the output schema. Let's avoid constrainting the output schema to only be an object. The most common output schema is a string. Free form jsonschema is useful here, we could constraint to omit $ref and $defs for simplicity.

Reply:
I think that is already supported:

Schema
```jsonc
{
  // ...
  "output_schema": {
    "type": "string"
    "description": "This tool returns a string"
  }
}
```

Call tool response
```jsonc
{
  call_id: "123e4567-e89b-12d3-a456-426614174000",
  success: true,
  value: "Hello world",
}
```

Reply:
yeah OK this makes sense -- an extra level of abstraction

Reply:
Schemas describing schemas, it's schemas all the way down 😉